django/db/models/sql/compiler.py

```diff
@@ -1793,6 +1808,13 @@ def as_sql(self):
                 result.append("VALUES (%s)" % ", ".join(placeholder_rows[0]))
                 params = [param_rows[0]]
             if on_conflict_suffix_sql:
+                (
+                    condition_sql,
+                    condition_params,
+                ) = self.compile_bulk_operation_condition_to_sql()
+                if condition_sql:
+                    on_conflict_suffix_sql += f" WHERE {condition_sql}"
+                    param_rows += [condition_params]
```

Comment:
Should this be `params += ...` here and not `param_rows += ...` 🤔 

There are two more `if on_conflict_suffix_sql:` branches below where this should also be added...

...although to avoid duplication perhaps we can just shift this up so it is just below where `on_conflict_suffix_sql` is defined?

Maybe something like the following if it works? It might well need tweaking...

```python
        on_conflict_suffix_sql = self.connection.ops.on_conflict_suffix_sql(
            ...
        )
        if on_conflict_suffix_sql:
            extra_sql, extra_params = self.compile_bulk_operation_condition_to_sql()
            if extra_sql:
                on_conflict_suffix_sql += f" WHERE {extra_sql}"
                param_rows += [extra_params]
```